src/main/clojure/clara/rules/platform.cljc

```diff
@@ -8,6 +8,22 @@
   (throw #?(:clj (IllegalArgumentException. description) :cljs (js/Error. description))))
 
 #?(:clj
+   (defn group-by-seq [f coll]
+     (let [^java.util.Map m (reduce (fn [^java.util.Map m x]
+                                      (let [k (f x)
+                                            xs (or (.get m k)
+                                                   (transient []))]
+                                        (.put m k (conj! xs x)))
+                                      m)
+                                    (java.util.LinkedHashMap.)
+                                    coll)]
+       (into []
+             (map (fn [^java.util.Map$Entry e] [(.getKey e) (persistent! (.getValue e))]))
```

Comment:
The issue you are referring to was specifically around IdentityHashMap, which had a difference in Java 6 vs 8.
In Java 6 it's Map$Entry object was actually mutated in place every time Iterator.next() was called, which doesn't work correctly if the entry set is iterated via `reduce` or `transduce`, which is harder to avoid.

However, since these iterators have changed fairly often in Java for these classes, it would be safer to just do our own loop-recur here.  I'm changing it.